llvm/lib/Target/X86/lab3/kruglov_alexey/X86KruglovInstCounterPass.cpp

```diff
@@ -0,0 +1,38 @@
+#include "X86.h"
+#include "X86InstrInfo.h"
+#include "X86Subtarget.h"
+#include "llvm/CodeGen/MachineFunctionPass.h"
+#include "llvm/CodeGen/MachineInstrBuilder.h"
+
+using namespace llvm;
+
+namespace {
+class X86KruglovCntPass : public MachineFunctionPass {
+public:
+  static inline char ID = 0;
+
+  X86KruglovCntPass() : MachineFunctionPass(ID) {}
+
+  bool runOnMachineFunction(MachineFunction &MF) override {
+    DebugLoc DebugLocation = MF.front().begin()->getDebugLoc();
+    const TargetInstrInfo *InstrInfo = MF.getSubtarget().getInstrInfo();
+
+    for (auto &MBB : MF) {
+      size_t Count = 0;
+      for (auto &MI : MBB) {
+        Count++;
+      }
+
+      BuildMI(MBB, MBB.getFirstTerminator(), DebugLocation,
+              InstrInfo->get(X86::ADD64ri32))
+          .addImm(Count)
+          .addExternalSymbol("ic");
```

Comment:
`addExternalSymbol` is for objects placed in external memory (for example, call to a function that is defined in another file to be linked with the current one later). On the other hand, `ic` is defined in global memory of current file.